internal/state/change_processor_test.go

```diff
@@ -613,38 +746,35 @@ var _ = Describe("ChangeProcessor", func() {
 						Conditions: conditions.NewGatewayInvalid("GatewayClass doesn't exist"),
 					}
 					expGraph.Routes = map[types.NamespacedName]*graph.Route{}
+					expGraph.Secrets = map[types.NamespacedName]*graph.Secret{}
 
 					changed, graphCfg := processor.Process()
 					Expect(changed).To(BeTrue())
 					Expect(helpers.Diff(expGraph, graphCfg)).To(BeEmpty())
 				})
 			})
 			When("the second Gateway is deleted", func() {
-				It("returns updated graph", func() {
+				It("returns empty graph", func() {
 					processor.CaptureDeleteChange(
 						&v1beta1.Gateway{},
 						types.NamespacedName{Namespace: "test", Name: "gateway-2"},
 					)
 
-					expGraph := &graph.Graph{}
-
 					changed, graphCfg := processor.Process()
 					Expect(changed).To(BeTrue())
-					Expect(helpers.Diff(expGraph, graphCfg)).To(BeEmpty())
```

Comment:
I think helpers.Diff actually gives better output when the graphs don't equal, right? It only shows the differing sections, whereas I think ExpectToEqual shows the entire contents, making it harder to parse.

Reply:
the reason I chanced because It was comparing it to the empty graph, so I expect in case of the error, all set fields in the result graph are incorrect. Below is an example of the error output:
```
[FAILED] Expected
      <*graph.Graph | 0xc000030660>: {
          GatewayClass: nil,
          Gateway: nil,
          IgnoredGatewayClasses: nil,
          IgnoredGateways: nil,
          Routes: {
              {Namespace: "test", Name: "hr-1"}: {
                  Source: {
                      TypeMeta: {Kind: "", APIVersion: ""},
                      ObjectMeta: {
                          Name: "hr-1",
                          GenerateName: "",
                          Namespace: "test",
                          SelfLink: "",
                          UID: "",
                          ResourceVersion: "",
                          Generation: 2,
                          CreationTimestamp: {
                              Time: 0001-01-01T00:00:00Z,
                          },
                          DeletionTimestamp: nil,
                          DeletionGracePeriodSeconds: nil,
                          Labels: nil,
                          Annotations: nil,
                          OwnerReferences: nil,
                          Finalizers: nil,
                          ManagedFields: nil,
                      },
                      Spec: {
                          CommonRouteSpec: {
                              ParentRefs: [
                                  {
                                      Group: nil,
                                      Kind: nil,
                                      Namespace: "test",
                                      Name: "gateway-1",
                                      SectionName: "listener-80-1",
                                      Port: nil,
                                  },
                                  {
                                      Group: nil,
                                      Kind: nil,
                                      Namespace: "test",
                                      Name: "gateway-1",
                                      SectionName: "listener-443-1",
                                      Port: nil,
                                  },
                              ],
                          },
                          Hostnames: ["foo.example.com"],
                          Rules: [
                              {
                                  Matches: [
                                      {
                                          Path: {Type: ..., Value: ...},
                                          Headers: nil,
                                          QueryParams: nil,
                                          Method: nil,
                                      },
                                  ],
                                  Filters: nil,
                                  BackendRefs: nil,
                              },
                          ],
                      },
                      Status: {
                          RouteStatus: {Parents: nil},
                      },
                  },
                  ParentRefs: nil,
                  Conditions: nil,
                  Rules: nil,
                  Valid: false,
              },
          },
          Secrets: {},
      }
  to equal
      <*graph.Graph | 0xc0000304e0>: {
          GatewayClass: nil,
          Gateway: nil,
          IgnoredGatewayClasses: nil,
          IgnoredGateways: nil,
          Routes: nil,
          Secrets: {},
```

and with helpers.Diff

```
  [FAILED] Expected
      <string>: (-want +got)
        &graph.Graph{
        	... // 2 identical fields
        	IgnoredGatewayClasses: nil,
        	IgnoredGateways:       nil,
      - 	Routes:                nil,
      + 	Routes: map[types.NamespacedName]*graph.Route{
      + 		s"test/hr-1": &{
      + 			Source: &v1beta1.HTTPRoute{ObjectMeta: v1.ObjectMeta{...}, Spec: v1beta1.HTTPRouteSpec{...}},
      + 		},
      + 	},
        	Secrets: {},
        }
      
  to be empty
```

with helpers.Diff it's still better and compact, so I will revert
👍